### PR TITLE
[WIP] Implementation: r_set_union()

### PR DESCRIPTION
**Pull request by matthiasbeyer**

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
@neithernut Please review. I rewrote the entire history, keeping (of course) you as Author of your commits (while squashing other commits into them in some cases).

File: src/libreset/ht/ht_union.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+int
+ht_union(
+    struct ht* dest,
+    struct ht const* src,
+    struct r_set_cfg const* cfg
+) {
+    size_t dest_b = ht_nbuckets(dest);
+    size_t bucket_ration = ht_nbuckets(src) / dest_b;
+    //!< @todo: implement case where bucket_ratio is 0
```
Comment (neithernut) [inline src/libreset/ht/ht_union.c:11]: If we merge with this TODO still in place, we should file an issue right away (so we don't forget about this).

Comment (matthiasbeyer) [inline src/libreset/ht/ht_union.c:11]: I don't want to merge right now, I cleaned the history and now you're welcome to implement this (last?) todo.

File: src/libreset/ht/ht_union.c [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#include "ht/ht.h"
+
+int
+ht_union(
+    struct ht* dest,
+    struct ht const* src,
+    struct r_set_cfg const* cfg
+) {
+    size_t dest_b = ht_nbuckets(dest);
+    size_t bucket_ration = ht_nbuckets(src) / dest_b;
```
Comment (TheNeikos) [inline src/libreset/ht/ht_union.c:10]: There is a typo in the variable name, we are talking mathematics not food :)

**Comment by matthiasbeyer** [comment]
@neithernut Do you have an overview if this is ready?

**Comment by neithernut** [comment]
Weekend.

File: src/libreset/ht/ht_union.c (hunk 2 above)
Comment (matthiasbeyer) [inline src/libreset/ht/ht_union.c:10]: :-)

**Comment by neithernut** [comment]
I'm done with the changes, but they are not yet tested. Please have a look at them. (branch in my repo)
I ended up with a different approach then intended originally.

**Comment by matthiasbeyer** [comment]
I guess a PR onto this branch would be sufficient. There is a lot of documentation missing, as I do not understand most of your changes and also not the approach.

Feel free to add documentation and notifying me again, without PR.

**Comment by neithernut** [comment]
The changes need to be squashed anyways, so a PR would be on this branch anyway.

**Comment by TheNeikos** [comment]
Could you also please fix the typo I pointed out!

**Comment by matthiasbeyer** [comment]
Somehow this PR broke by moving the repository. I'll resubmit this in another PR.

**Pull request closed without merging**